ush/compare_f90nml.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+
+import json
+from typing import Dict
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+try:
+    import f90nml
+except (ImportError, ModuleNotFoundError):
+    raise ModuleNotFoundError(f"f90nml not found on this system, ABORT!")
+
+
+def get_dict_from_nml(filename: str) -> Dict:
+    """
+    Read a F90 namelist and convert to a dictionary.
+    This method uses json to convert OrderedDictionary into regular dictionary
+    """
+    return json.loads(json.dumps(f90nml.read(filename).todict()))
+
+
+def compare_dicts(dict1: Dict, dict2: Dict, path: str = "") -> None:
+    """
+    Compare 2 dictionaries.
+    This is done by looping over keys in dictionary 1 and searching for them
+    in dictionary 2.
+    If a matching key is found, the values are compared.
+    If a matching key is not found, it is set to as UNDEFINED.
+    Note: A reverse match is not performed in this method.  For reverse matching, use the -r option in the main driver.
```

Comment:
Might be a future improvement, but I think this would be even more useful if it inherently did both directions. As-is, there is a need to run it twice and mentally ignore duplicates.

Possible implementation:
Move all the diffing to another hidden function (like print), call it twice, and return the results dict. Then reverse all the lists in the second one and combine the dicts before sending it to the print function.

Reply:
Deferred to later.